src/dashboard/src/components/archival_storage/tests/test_archival_storage.py

```diff
@@ -200,6 +204,154 @@ def test_get_pointer_known_pointer(
+            "encrypted": True,
+            "AICID": None,
+            "countAIPsinAIC": None,
+            "accessionids": ["Àà", "Éé", "Îî", "Ôô", "Ùù"],
+            "uuid": "22423d5c-f992-4979-9390-1cb61c87da14",
+            "name": "tz",
+            "created": 1594938200,
+            "file_count": 2,
+            "location": "thé cloud",
+            "isPartOf": None,
+            "size": "152.1\xa0KB",
+            "type": "AIP",
+        },
+    ]
+    mocker.patch("elasticSearchFunctions.get_client")
+    mocker.patch("elasticsearch.Elasticsearch.search")
+    mocker.patch(
+        "components.archival_storage.views.search_augment_aip_results",
+        return_value=mock_augmented_result,
+    )
+    REQUEST_PARAMS = {
+        "requestFile": True,
+        "mimeType": CSV_MIME,
+        "fileName": RESULT_FILENAME,
+        "returnAll": True,
+    }
+    response = client.get(
+        "/archival-storage/search/?{}".format(urlencode(REQUEST_PARAMS))
+    )
+
+    # Check that our response headers are going to be useful to the caller.
+    assert response.get("content-type") == CSV_MIME
+    assert response.get("content-disposition") == RESULT_DISPOSITION
+
+    streamed_content = b"".join([content for content in response.streaming_content])
+    csv_file = StringIO(streamed_content)
+    data_frame = pd.read_csv(csv_file, header=0, encoding="utf8")
+
+    # Make sure that our headers come out as expected and in the right order.
+    assert list(data_frame.columns.values) == ordered_headers
+
+    # Make some assertions about the quality of the rest of our data,
+    # particularly those we're encoding as
+    accession_ids = data_frame["Accession IDs"]
+    assert pd.isnull(accession_ids[0])
+    assert accession_ids[1] == "; ".join(mock_augmented_result[1].get("accessionids"))
+
+    encyypted = data_frame["Encrypted"]
+    assert encyypted[0] == mock_augmented_result[0].get("encrypted")
+    assert encyypted[1] == mock_augmented_result[1].get("encrypted")
+
+    size = data_frame["Size"]
+    assert size[0] == mock_augmented_result[0].get("size")
+    assert size[1] == mock_augmented_result[1].get("size")
+
+    location = data_frame["Location"]
+    assert location[0] == mock_augmented_result[0].get("location")
+    assert location[1] == mock_augmented_result[1].get("location")
+
+
+def test_search_as_csv_invalid_route(
+    client, mocker, django_user_model, username, password, tmp_path
+):
+    """Test search as CSV invalid rute
+
+    Given the ability to download the Elasticsearch AIP index table as a
+    CSV, make sure that the new pathway through the code does not
+    impact the regular Archival Search component by performing a
+    rudimentary test to just ensure that we can successfully NOT
+    download the results as a CSV.
+    """
+    dashboard_login_and_setup(client, django_user_model, username, password)
+    MOCK_TOTAL = 10
+    AUG_RESULTS = {"mock": "response"}
+    CONTENT_TYPE_HDR = "content-type"
```

Comment:
Feel free to ignore, but `CONTENT_TYPE_HDR` and `JSON_MIME` could be a constants at file scope, since they are used in a few different tests.

Reply:
Good call. And indeed, `CONTENT_TYPE` had already been used at file scope so I was just repeating myself adding it again here and without looking there previously.